src/providers/MeetingProvider/MeetingManager.ts

```diff
@@ -134,7 +138,7 @@ export class MeetingManager implements AudioVideoObserver {
     this.audioVideoObservers = {};
   }
 
-  async join({ meetingInfo, attendeeInfo }: MeetingJoinData) {
+  async join({ meetingInfo, attendeeInfo, deviceLabels = DeviceLabels.AudioAndVideo }: MeetingJoinData) {
```

Comment:
Can we not use join for this? DeviceLabel is not join data so its weird to include them here.

Reply:
Should I move the `deviceLabels` out of the `MeetingInfo` but still keep it as an optional parameter in `join` function? Or should I have a function `setDeviceLabels` to change  the `deviceLabels` in `meetingManager`? I would prefer the first solution.

Reply:
The problem is that it's not really suitable as a constructor argument (both because you might not know the value at that time, and also it's awkward to thread a new `MeetingManager` into `MeetingProvider`), and mutating state sucks.

It's not join data, but it's a very clear description of what the builder intends the join to be: with audio or without.

Reply:
Keep the current solution.